### PR TITLE
feat: regularize address

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/AddressRegistry.Api.BackOffice/Validators/ValidationErrorCodes.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public static class ValidationErrorCodes
         public const string AddressCannotBeApproved = "AdresGehistoreerdOfAfgekeurd";
         public const string AddressCannotBeRejected = "AdresGehistoreerdOfGoedgekeurd";
         public const string AddressCannotBeDeregulated = "AdresGehistoreerdOfAfgekeurd";
+        public const string AddressCannotBeRegularized = "AdresGehistoreerdOfAfgekeurd";
```
Comment (lvermeulen) [inline src/AddressRegistry.Api.BackOffice/Validators/ValidationErrorCodes.cs:21]: is this correct? (same as AddressCannotBeDeregulated)

Comment (emalfroy) [reply src/AddressRegistry.Api.BackOffice/Validators/ValidationErrorCodes.cs:21]: Yes, see ticket 3362
```
Een POST op {{adressen/<id>/acties/regulariseren}} voor een adres met status 
afgekeurd OF gehistoreerd 
Geeft als resultaat 400
```

The error code is in line with the other error codes

File: test/AddressRegistry.Tests/AggregateTests/WhenRegularizingAddress/GivenStreetNameExists.cs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                    addressWasDeregulated)
+                .When(command)
+                .Then(new Fact(_streamId,
+                    new AddressWasRegularized(
+                        Fixture.Create<StreetNamePersistentLocalId>(),
+                        addressPersistentLocalId))));
+        }
+
+        [Fact]
+        public void WithoutProposedAddress_ThenThrow()
```
Comment (ArneD) [inline test/AddressRegistry.Tests/AggregateTests/WhenRegularizingAddress/GivenStreetNameExists.cs:67]: `ThenThrowsxxx` specify what it throws. Check whole file.

File: test/AddressRegistry.Tests/BackOffice/Api/WhenRegularizingAddress/GivenAddressDoesNotExist.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        private readonly TestBackOfficeContext _backOfficeContext;
+
+        public GivenAddressDoesNotExist(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            _controller = CreateApiBusControllerWithUser<AddressController>();
+            _backOfficeContext = new FakeBackOfficeContextFactory().CreateDbContext();
+        }
+
+        [Fact]
+        public void ThenThrowValidationException()
```
Comment (ArneD) [inline test/AddressRegistry.Tests/BackOffice/Api/WhenRegularizingAddress/GivenAddressDoesNotExist.cs:35]: `ThenThrowsxxx`.

File: test/AddressRegistry.Tests/BackOffice/Api/WhenRegularizingAddress/GivenAddressHasInvalidStatus.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        public GivenAddressHasInvalidStatus(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            _controller = CreateApiBusControllerWithUser<AddressController>();
+            _idempotencyContext = new FakeIdempotencyContextFactory().CreateDbContext();
+            _backOfficeContext = new FakeBackOfficeContextFactory().CreateDbContext();
+        }
+
+        [Fact]
+        public void ThenThrowApiException()
```
Comment (ArneD) [inline test/AddressRegistry.Tests/BackOffice/Api/WhenRegularizingAddress/GivenAddressHasInvalidStatus.cs:37]: `ThenThrowsxxx`

**Review by ArneD — COMMENTED** [review]
Missing list projection + test => timestamp change.

**Pull request merged**